sklego/preprocessing.py

```diff
@@ -144,3 +144,52 @@ def _check_X_for_type(X):
         """Checks if input of the Selector is of the required dtype"""
         if not isinstance(X, pd.DataFrame):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
+            
+            
+            
+
+class ColumnsCapper(TransformerMixin):
+    """Returns a dataframe with selected columns capped to a maximum threshold.
+    the threshold can be set manually or a quantile can be used.
+    
+    param:
+    X: `pd.DataFrame` to which to apply capping
+    quantile_thresholds: `float or list of floats` if float (e.g. 0.9) is the quantile to use as threshold. if 
+                            list, there is one value for each column to cap
+    absolute_threshold: `float or list of floats` (e.g. 100) absolute value to use to cap values. if list, there is one
+                            value for each column to cap
+    cols_names: `list of column names or None`. if list, cap these columns. if None, caps all columns in dataframe
+    use_quantile: `bool`. if true uses the quantile, otherwise uses the absolute threshold
+    
+    
+    """
+    
+    def __init__(self, quantile_thresholds, cols_names=None, absolute_thresholds=None, use_quantile=True):
```

Comment:
after thinking about it... what is a more obvious api? 

```python
ColumnCapper(col_names=["foo", "bar"], threshold=[0.6, 0.1], method="absolute")
```

vs. 

```python
ColumnCapper(col_names=["foo", "bar"], 
                           absolute_thresholds =[0.6, 0.1])
```

@MBrouns do you have an opinion on this? It feels like we might experience this issue a bunch of times in our APIs.

Reply:
In the second case I kind of expect to be able to set multiple thresholds (one for each type), if that is the case I would go with the second, otherwise with the first